### PR TITLE
Time reported is incorrect

### PR DESCRIPTION
**Pull request by Schramp**
I noticed the time reported by the tool is significantly off

### CODE REVIEW

**Comment by ydkhatri** [comment]
Thanks for submitting these, I'll review and update soon.

**Comment by Schramp** [comment]
Goodluck

**Comment by ydkhatri** [comment]
Thanks for the fixes. It might be a while before I get to them though.

**Comment by joachimmetz** [comment]
@Schramp do you need a review on this?

File: .github/workflows/test_docker.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      image: registry.fedoraproject.org/fedora:${{ matrix.version }}
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up container
+      run: |
+        dnf install -y dnf-plugins-core langpacks-en
+    - name: Install dependencies
+      run: |
+        dnf copr -y enable @gift/dev
+        dnf install -y @development-tools python3 python3-devel libfwps-python3 libfwsi-python3 libgzipf-python3 liblnk-python3 libolecf-python3 libregf-python3 python3-dfdatetime python3-dtfabric python3-lz4 python3-pyyaml python3-setuptools
```
Comment (joachimmetz) [inline .github/workflows/test_docker.yml:21]: remove the libyal specific dependencies, I assume this was copied from dtformats, that use these but not this project AFAIK

File: .github/workflows/test_docker.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      run: |
+        python3 setup.py install
+    - name: Run tests
+      env:
+        LANG: C.utf8
+      run: |
+        python3 ./run_tests.py
+    - name: Run end-to-end tests
+      run: |
+        if test -f tests/end-to-end.py; then PYTHONPATH=. python3 ./tests/end-to-end.py --debug -c config/end-to-end.ini; fi
```
Comment (joachimmetz) [inline .github/workflows/test_docker.yml:34]: Don't think this project comes with `tests/end-to-end.py`, remove?

File: .github/workflows/test_tox.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        apt-get install -y libterm-readline-gnu-perl locales software-properties-common
+        locale-gen en_US.UTF-8
+        ln -f -s /usr/share/zoneinfo/UTC /etc/localtime
+    - name: Install dependencies
+      env:
+        DEBIAN_FRONTEND: noninteractive
+      run: |
+        add-apt-repository -y universe
+        add-apt-repository -y ppa:deadsnakes/ppa
+        add-apt-repository -y ppa:gift/dev
```
Comment (joachimmetz) [inline .github/workflows/test_tox.yml:45]: You might not need GIFT PPA if you remove the unused dependencies

File: UnifiedLog/Lib.py [hunk 4]
```diff
@@ -58,6 +58,11 @@ def ReadAPFSTime(mac_apfs_time): # Mac APFS timestamp is nano second time epoch
             if type(mac_apfs_time) == str:
                 mac_apfs_time = float(mac_apfs_time)
             return datetime.datetime(1970, 1, 1) + datetime.timedelta(seconds=mac_apfs_time/1000000000.)
```
Comment (joachimmetz) [inline UnifiedLog/Lib.py:60]: Why not use a time delta in `microseconds` for higher precision?

File: UnifiedLog/tracev3_file.py [hunk 5]
```diff
@@ -396,8 +902,8 @@ def CreateLossMsg(self, ts, start_ct, ct_base, buffer, buf_size):
             msg = 'lost {}{} unreliable messages from {} - {}  (exact start-approx. end)'
             sign, end_ct_rel, count = struct.unpack('<IQI', buffer[0:16])
             end_ct = ct_base + end_ct_rel
-            end_time = ts.time_stamp + end_ct - ts.continuousTime
-            start_time = ts.time_stamp + start_ct - ts.continuousTime
+            end_time = ts.time_stamp + (end_ct - ts.continuousTime)*(1.0*ts.ts_numerator)/ts.ts_denominator
```
Comment (joachimmetz) [inline UnifiedLog/tracev3_file.py:905]: why the `1.0*` ? the result of `/` on Python 3 will be a float

**Comment by Schramp** [comment]
Thanks for this feedback, it mostly allready on my main, so I will try to change it there.

My intention is to make my fork the base for all future devs, however github as platform is complicating things a bit. They seem to globaly define a master fork on which the PR's and incidents are registered. At the same time @ydkhatri needs to get all the credit for his work, so it would be weird to transfer ownership of his repo, besides that, it will break all references on the internet to @ydkhatri work.

The technical trick which probably solves it is if @ydkhatri transfers ownership of this repo to my account, and fork back on his account or perhaps make a local fork of his repo within his account, transfer the base and rename the fork to the old name.

I haven't tried to contact github themselves yet if they can do something equivalent from their backend.. Does anyone of you have experience with it?

File: .github/workflows/test_docker.yml (hunk 1 above)
Comment (Schramp) [reply .github/workflows/test_docker.yml:21]: Yes, I did. I liked the way you incorporated code coverage as part of my PR. Compliments.

**Comment by joachimmetz** [comment]
> Thanks for this feedback, it mostly allready on my main, so I will try to change it there.

sounds good

> I haven't tried to contact github themselves yet if they can do something equivalent from their backend.. Does anyone of you have experience with it?

sounds that this might be easier via @ydkhatri 

I would recommend setting up a separate github org and transferring the repo there.

File: .github/workflows/test_docker.yml (hunk 1 above)
Comment (joachimmetz) [reply .github/workflows/test_docker.yml:21]: > Yes, I did. I liked the way you incorporated code coverage as part of my PR. Compliments.

this is functionality codecov.io provides

**Comment by Schramp** [comment]
Closing the PR here.

**Pull request closed without merging**